### PR TITLE
Shields rework: generators, regenerators, and capacitors

### PR DESCRIPTION
**Pull request by Petersupes**
**Content (Artwork / Balance)**

This PR addresses the features described in issues #9460, #503 

## Summary
Two large problems within ES are the lack of incentives to play as a smaller ship, and what seem at times to be unfair too quick deaths with no way to avoid this other than getting a bigger ship, compounding the first problem. An idea to alleviate these issues is adding regenerators and capacitors. Regenerators are much more powerful than standard generators, but have shield delay, preventing them from being used to face tank like HWs do. Capacitors increase the shields of a ship slightly while costing shield energy, making them viable for all ships but in particular for fighters/drones who can rely entirely on their carrier to recharge shields, and ignore the penalty of the increased regeneration cost while in combat while allowing them to be able to absorb more than 1 stray missile before exploding. Like capacitors, plating increases the hull of the ship, while adding excessive mass and cooling inefficiency. In addition to these new outfit types, all standard generators have been increased in shield generation by 50% to address the problem of deaths that seem too quick. Shield energy has not been touched, so all previous energy setups would work the same with regular shield generators.

Generators:
  - Increased shield generation/hull repair by 50% for all outfits to decrease time to kill

Regenerators:
  - The syndicate and hai shield generators have had both their generation and energy increased by 120%
  - 2 regenerators from lovelace labs available at the start of the game to provide the option early game when a typical player will have a smaller ship
  - An additional syndicate regenerator that becomes available at the same time as the barb, expanding options and narrative-wise shows that the syndicate are developing the pre-existing regenerators
  - Regenerators have large shield delays requiring ships to leave combat to reshield

Capacitors:
  - 4 human capacitors and 2 hai capacitors
  - 2 human hull platings

Spreadsheet detailing old and new outfits: https://docs.google.com/spreadsheets/d/1cv67CbQcK1RFmtotjBOPiZsvutAwC4Al5ox6FNmMXDo/edit?usp=sharing

## Potential Issues
A reason against capacitors brought up time and again is the fact that by providing outfits that allow players to increase the shields/hull of a ship is that it can create balancing issues wherein large cargo ships could become incredibly tanky and powerful through outfits expansions. However, the cooling inefficiency on outfits expansions already soft prohibits this as any amount of outfit expansion capable of radically increasing the total health of a ship would cripple it through any kind of heat based interaction. I believe that this, coupled with the drawbacks of the capacitors & plating gives them the potential to be balanced and open up more interesting play styles. If this is deemed insufficient for balance, an amount of depleted delay could be added to shield capacitors.

Secondly with the change of the syndicate and hai regenerators, some marauders and all hai ships will no longer reshield under fire. This is more of an issue for the latter as it significantly changes both the hai reveal and wanderer campaigns in terms of these ships. A new ship personality would probably be needed to prevent hai ships from attempting to face tank when they can't, as well as if regenerators are used in other fleets.

## Examples
A bastion has 7500 shields and comes equipped with a D67-TM Shield generator. By installing the largest human capacitor it would gain ~50% more shields while doubling its shield generation cost.

A lance equipping 2 of the smallest human capacitors in place of its shield generator would more than double its shield while freeing up energy, but be unable to reshield itself.

The changes to regular shield generators mean that a D67 is able to fully block a single beam laser, and a D94 will be able to fully block a single heavy laser.

A hawk with its stock generator, D23, would take 86.8 seconds to fully charge after the 50% increase is applied, this makes it both boring, tedious and at times impossible to fight larger ships with an interceptor. With a comparably sized L-360 regenerator, it would fully charge in 30.5 seconds, including the 3.5 seconds needed without taking any damage, allowing a hawk to relatively quickly disengage from and reenter combat and challenge larger ships.

## Notes
The remnant's built in shield generation and hull repair are currently untouched.

## To Do
 - [ ] Outfit descriptions
 - [ ] Outfit sprites
 - [ ] More variety of alien regenerators + capacitors (optional)
## Artwork Checklist
 - [ ] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified
 - [ ] I created a PR to the [endless-sky-assets repo](https://github.com/endless-sky/endless-sky-assets) with the necessary image, blend, and texture assets:
 - [ ] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets:

### CODE REVIEW

File: data/human/outfits.txt [hunk 1]
```diff
@@ -105,24 +105,122 @@ outfit "S-270 Regenerator"
 	thumbnail "outfit/small regenerator"
 	"mass" 12
 	"outfit space" -12
-	"shield generation" .6
-	"shield energy" .6
+	"shield generation" 1.32
+	"shield energy" 1.32
```
Comment (Quantumshark) [inline data/human/outfits.txt:109]: I presume you meant to give this and the S-970 a shield delay? This is more than the 50% increase you mentioned.

**Comment by Quantumshark** [comment]
>In addition to these new outfit types, all standard generators have been increased in shield generation by 50% to address the problem of deaths that seem too quick.

Cases where you die very quickly are going to be the _least_ affected by that change. Shield generation has the least effect in the shortest fights. This change will have a lot of effects in every other area, but won't change much toward your desired goal. That's not to say it's necessarily a bad idea, but it's not a good way of extending survival in short time-to-kill scenarios.

**Comment by Petersupes** [comment]
I see what you're saying, most quick deaths do tend to be from burst damage sources like missiles, which the capacitors aim to alleviate. But I still think its a decent change as currently ships taking ~2 minutes to fully shield isn't really fun gameplay.

**Comment by Zitchas** [comment]
Overall a good idea, however I disagree with the general buff to rate it recharge. Firstly, as Warpcore mentioned, that has the least benefit in fast fights, so didn't really help the problem.
Secondly, it helps tender tactical combat more useless, and increases the emphasis on "must use biggest ship with biggest DPS possible" build, since agility and strike-and-run become even more useless.

Along with these changes, we should probably be swapping load outs so that slow and tanky ships don't have regenerators, while small and agile ships tend to do so, and the sizes and power of the regenerators should be adjusted to match those ships.

**Comment by Quantumshark** [comment]
>Secondly, it helps tender tactical combat more useless, and increases the emphasis on "must use biggest ship with biggest DPS possible" build, since agility and strike-and-run become even more useless.

I'd say that's not entirely true.
With a hit-and-run tactical approach, what you're doing is pitting your regen and damage against the target's regen and damage - the condition for victory through such an approach is that it takes less time for you to recover from your target's attacks than for them to recover from yours. This global increase to regen affects both sides evenly, so there's no overall change - the strategy is neither more nor less viable/effective in any given circumstance. So, it doesn't help with the goal of this PR - that being, to make hit-and-run tactics more viable - but it doesn't make them any less viable either.

**Comment by Petersupes** [comment]
> I don't think this is the case. Increasing regen globally (and not increasing weapons DPS to match) diminishes the effect of an individual attack, spreading out the conditions for victory across the total length of the battle. To use an extreme example, a fight where each side could kill the other in a single hit would involve much more reliance on skill in aiming and dodging (and luck!) compared to a prolonged brawl with a low damage weapon, where simply kiting or rotation-locking the AI would be the only way to ensure victory.

The global regen increase doesn't affect that much, weapons were already far stronger than shield generators with a single heavy laser out-damaging a D94 generator, opposite the purpose of a sufficient generator which should be to block small weapons during combat. And while yes i agree that too tedious a battle where the player has to resort to kiting are bad thing, I think that a battle won in a single shot can be far more frustrating to a player. So while ideally a perfect middle ground should be achieved, in practice that's nearly impossible to nail down and leaning slightly away from the more frustrating option is preferable.

**Comment by Anarchist2** [comment]
Regenerators are directly mentioned to differ from regular generators in that they continuously modify a shield's geometry, which is reflected by how they have a constant resource drain even when not charging. So why are they the generators that have been changed to use shield delay, making them only work some of the time?

**Comment by Amazinite** [comment]
We already have @Quantumshark working on a large mass rework with #9451/#9616. I think also having a regen rework open at the same time is jumping the gun. As noted, one of the potential reasons for reworking how regen works would be to change up the balance between face tanking your target and using hit and run tactics, but the mass changes will also change up that balance. Since both of these realms of balance are impacting the same gameplay elements, we should be tackling one before we tackle the other rather than trying to do both at once and potentially causing the compounded effect of both changes to swing things too far in the other direction.

**Pull request closed without merging**